spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/audit/service/DefaultAuditRecordService.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.server.audit.service;
+
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.springframework.cloud.dataflow.server.audit.domain.AuditActionType;
+import org.springframework.cloud.dataflow.server.audit.domain.AuditOperationType;
+import org.springframework.cloud.dataflow.server.audit.domain.AuditRecord;
+import org.springframework.cloud.dataflow.server.audit.repository.AuditRecordRepository;
+import org.springframework.util.Assert;
+
+
+/**
+ * Default implementation of the {@link AuditRecordService}.
+ *
+ * @author Gunnar Hillert
+ *
+ */
+public class DefaultAuditRecordService implements AuditRecordService {
+
+	private final AuditRecordRepository auditRecordRepository;
+
+	private final ObjectMapper objectMapper;
+
+	public DefaultAuditRecordService(AuditRecordRepository auditRecordRepository, ObjectMapper objectMapper) {
+		super();
+		Assert.notNull(auditRecordRepository, "auditRecordRepository must not be null.");
+		Assert.notNull(auditRecordRepository, "objectMapper must not be null.");
+		this.auditRecordRepository = auditRecordRepository;
+		this.objectMapper = objectMapper;
+	}
+
+	@Override
+	public void populateAndSaveAuditRecord(AuditActionType auditActionType, AuditOperationType auditOperationType,
```

Comment:
Small point, but having the operation type first, then the action type reads a bit better to me.

Instead of 
`AuditActionType.CREATE, AuditOperationType.STREAM`

use

`AuditOperationType.STREAM, AuditActionType.CREATE`